intentgen/src/main/java/com/chatbot/IntentGenerator.java

```diff
@@ -0,0 +1,190 @@
+package com.chatbot;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import com.chatbot.protobuf.UseCaseOuterClass.UseCase;
+import com.google.cloud.dialogflow.v2.Intent;
+import com.google.cloud.dialogflow.v2.Context;
+import com.google.cloud.dialogflow.v2.Intent.TrainingPhrase;
+import com.google.cloud.dialogflow.v2.Intent.TrainingPhrase.Part;
+import com.google.cloud.dialogflow.v2.IntentsClient;
+import com.google.cloud.dialogflow.v2.ProjectAgentName;
+import com.google.protobuf.TextFormat;
+
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.CommandLineParser;
+import org.apache.commons.cli.DefaultParser;
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.ParseException;
+
+public class IntentGenerator {
+  static final List<String> YES_TRAINING_PHRASES = new ArrayList<String>(Arrays.asList("yes",
+      "okay I will", "why not", "yes that's alright", "yes I do", "exactly", "of course",
+      "yep that's ok", "okay", "ok", "sure"));
+  static final List<String> NO_TRAINING_PHRASES = new ArrayList<String>(Arrays.asList("No",
+      "thanks but no", "no way", "no no don't", "na", "no it isn't", "don't", "nah I'm good",
+      "no I cannot", "I can't"));
+
+  public static void main(final String[] args) throws ParseException {
+    final String useCaseFile = getUseCaseFile(args);
+    final String projectID = System.getenv("projectID");
+    final ProjectAgentName parent = ProjectAgentName.of(projectID);
+    try(IntentsClient intentsClient = IntentsClient.create()) {
+      // this map is needed to check if an intent display name already exists and if so, instead of 
+      // creating an intent (which would return an error) we update the intent using the name
+      final Map<String, String> intentDisplayNameToName = getIntentDisplayNametoNameMapping(intentsClient,
+          parent);
+      final List<Intent> intentProtobufList = getIntentsInUseCase(useCaseFile).stream()
+          .map(x->intentToIntentProtobuf(x, intentDisplayNameToName, projectID))
+          .collect(Collectors.toList());
+      createOrUpdateIntents(intentProtobufList, intentsClient, intentDisplayNameToName, parent);
```

Comment:
Add test to check if both creation and updation are working as intended.

Reply:
Tests here would mean actually making calls to create/update intents, do we want tests to be changing the state of the agent?

Reply:
You can mock the things which are directly related to actual project and then test.
We would never want to change the state of the actual project via tests.

The current set of tests are not exhaustive enough, apart from the ones mentioned by Yogesh, try to add tests for failure cases as well. Eg. Input file is empty etc.